src/main/scala/cognite/spark/v1/PushdownUtilities.scala

```diff
@@ -18,7 +18,10 @@ import sttp.model.Uri
 
 import scala.util.Try
 
-final case class DeleteItem(id: Long)
+final case class DeleteItem(
+    id: Option[Long],
+    externalId: Option[String]
+)
```

Comment:
```suggestion
) {
   def toCogniteId: CogniteId = (id, externalId) match {
      case (None, None) => error
      case (Some(_), Some(_)) => error
      case (Some(id), None) => CogniteInternalId(id),
      case (None, Some(externalId)) => CogniteExternalId(externalId)
   }
}
```


I think this function will allow you to simplify the rest a bit ;)

Reply:
as discuss to avoid the conflict on id and externalId, I use this function but update it a bit more ;)